Cloud computing tutorial

abarciauskas-bgse opened the pull request:
ref: https://github.com/uwhackweek/schedule-2024/issues/35

abarciauskas-bgse committed:
Initial cloud computing tutorial notebooks

abarciauskas-bgse committed:
updates to notebooks

abarciauskas-bgse committed:
make data markdown

abarciauskas-bgse committed:
Revisions

abarciauskas-bgse commented:
I will look into the clean notebooks failure.

abarciauskas-bgse committed:
Clear executions

abarciauskas-bgse committed:
Fix spelling errors

abarciauskas-bgse committed:
Add tutorial to TOC

abarciauskas-bgse committed:
Try to fix TOC and add cloud computing to tutorials index

abarciauskas-bgse committed:
add to table of contents

abarciauskas-bgse committed:
Try to fix TOC

abarciauskas-bgse committed:
Try to fix TOC

abarciauskas-bgse committed:
Try to fix TOC

abarciauskas-bgse committed:
Fix some directives

abarciauskas-bgse committed:
Add file and try to hide it

abarciauskas-bgse committed:
Try using titles only

abarciauskas-bgse committed:
Skip execution

abarciauskas-bgse committed:
Use nb_execution_mode: off

abarciauskas-bgse committed:
Add nb_execution_modeoff

abarciauskas-bgse commented:
@scottyhq I'm having trouble getting cells _not_ to run in this tutorial. There are 2 notebooks with code cells:

1. book/tutorials/cloud-computing/04-cloud-optimized-icesat2.ipynb - this one should definitely be included in the navigation and tutorial but I don't think we want the cells to execute (it would be fine if we did but there is a geopandas import error, and other imports would probably fail too)
2. book/tutorials/cloud-computing/atl08_parquet_files/atl08_parquet.ipynb - This notebook should definitely _not_ be run as it includes code to generate geoparquet. Ideally this would not show up in the table of contents at all but just accessible via the link in the 04-cloud-optimized-icesat2.ipynb. But if I remove it from the TOC then the link in 04-cloud-optimized-icesat2.ipynb shows up as broken. As a fallback option, I could remove this NB from the TOC and just link to the github URL for that directory.

Let me know what you think and thanks in advance for your help.

scottyhq commented:
> this one should definitely be included in the navigation and tutorial but I don't think we want the cells to execute

The way we've approached this in the past is adding it's path unnder `exclude_patterns` in the config. Be sure to commit the *rendered* notebook if you do this
https://github.com/ICESAT-2HackWeek/website-2024/blob/db547fedb410bae8b7bc8a98cb1547b702662f44/book/_config.yml#L37-L39

> Ideally this would not show up in the table of contents at all but just accessible via the link in the 04-cloud-optimized-icesat2.ipynb.

Looks like you can use a `{download}` admonition. Not sure where this is documented, but the jupyterbook docs show this:
https://github.com/jupyter-book/jupyter-book/blob/bb13ee76f266ca0b94b31bc22e5480b60a52002d/docs/file-types/notebooks.ipynb#L11

abarciauskas-bgse committed:
Minor changes to text and formatting

abarciauskas-bgse commented:
I noticed the exclude pattern, but also that this line is commented out: https://github.com/ICESAT-2HackWeek/website-2024/blob/main/.github/workflows/ensure_clean_notebooks.py#L33 so I wasn't sure if I should use it. I'll try it with that line un-commented.

abarciauskas-bgse committed:
Exclude some cloud computing notebooks

scottyhq commented:
> /Users/runner/work/website-2024/website-2024/book/tutorials/cloud-computing/03-cloud-optimized-data-access.ipynb:10011: WARNING: local id not found in doc 'tutorials/cloud-computing/02-cloud-data-access': 'cloud-vs-local-storage' [myst.xref_missing]

I'm not sure why this link isn't working in your third notebook. Maybe it is sensitive to Caps or maybe it only works to do this within the same notebook... Feel free to try and fix or just link to the notebook without the section
```
[Cloud Data Access Notebook: Cloud vs Local Storage](./02-cloud-data-access.ipynb#cloud-vs-local-storage).
```

abarciauskas-bgse commented:
@scottyhq ~the link, when I try to use it in hub.cryointhecloud.com _is_ working which is why I am also confounded by this error.~ I was wrong it's not working, it just seemed like it was because of how I was working between notebooks 🤦🏽

abarciauskas-bgse committed:
Remove anchor link

scottyhq reviewed (approved):
Fantastic tutorial @abarciauskas-bgse thank you !'m on a plane back to seattle with limited wifi and the lonboard plot still worked well :)

scottyhq commented on book/tutorials/cloud-computing/atl08_parquet_files/atl08_parquet.ipynb:
```diff
@@ -0,0 +1,300 @@
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "dataset = pq.ParquetDataset(\"atl08_parquet\", partitioning=\"hive\", filters=[('year', '>=', 2021),\n",
```
optional: could add some notes on what `partitioning="hive"` is

scottyhq commented on book/tutorials/cloud-computing/atl08_parquet_files/atl08_parquet.ipynb:
```diff
@@ -0,0 +1,300 @@
+    "df['geometry'] = df['geometry'].apply(wkb.loads)\n",
+    "\n",
+    "\n",
+    "gdf = gpd.GeoDataFrame(df, geometry='geometry')\n",
```
```suggestion
    "gdf = gpd.GeoDataFrame(df, geometry='geometry', crs='EPSG:4326')\n",
```

(to avoid UserWarning: No CRS exists on data. from lonboard)

scottyhq commented on book/tutorials/cloud-computing/04-cloud-optimized-icesat2.ipynb:
```diff
@@ -0,0 +1,147 @@
+    "h_canopy = gdf_filtered['h_canopy']\n",
+    "h_canopy_normalized = (h_canopy - min_bound) / (max_bound - min_bound)\n",
+    "\n",
+    "layer = ScatterplotLayer.from_geopandas(gdf_filtered, radius_min_pixels=0.5)\n",
```
optional: note on how radius_min_pixels works (I noticed that zooming in on the map points get very small)

abarciauskas-bgse committed:
Add/modify images in 3rd notebook and add introdutory text

abarciauskas-bgse committed:
Add images

abarciauskas-bgse committed:
Address Scott's comments

abarciauskas-bgse commented:
Thanks @scottyhq so much for your review ❤️ I have addressed your comments and made a few more minor changes (added 4 introductory sentences to the 00 notebook, modified one image and added one image to the 03 notebook), but given these are minor changes I am going to merge. 

I am hoping to create a bonus notebook with more cloud computing resources 🤞🏽 if I have time.
